packages/client/src/name.js

```diff
@@ -0,0 +1,120 @@
+import { fromString as uint8ArrayFromString } from 'uint8arrays/from-string'
+import { toString as uint8ArrayToString } from 'uint8arrays/to-string'
+import { equals as uint8ArrayEquals } from 'uint8arrays/equals'
+import * as ipns from 'ipns'
+import * as Digest from 'multiformats/hashes/digest'
+import { identity } from 'multiformats/hashes/identity'
+import { base36 } from 'multiformats/bases/base36'
+import { CID } from 'multiformats/cid'
+import { keys } from 'libp2p-crypto'
+import { fetch } from './platform.js'
+
+const libp2pKeyCode = 0x72
+
+/**
+ * Create a private key, and key ID (the "name" used to resolve the current
+ * value).
+ */
+export async function keypair () {
+  const privKeyObj = await keys.generateKeyPair('Ed25519', 2048)
+  const digest = Digest.create(identity.code, privKeyObj.public.bytes)
+  return {
+    id: CID.createV1(libp2pKeyCode, digest).toString(base36),
+    privateKey: uint8ArrayToString(privKeyObj.bytes, 'base64pad')
+  }
+}
+
+/**
+ * Create a new IPNS record.
+ *
+ * @param {string} privKey Base 64 encoded private key bytes.
+ * @param {string|null} record Base 64 encoded bytes of the existing IPNS
+ * record. Records are created with a sequence number incremented from the
+ * previous value. Expected to be `null` when creating a brand new record.
+ * @param {string} value IPFS path.
+ */
+export async function create (privKey, record, value) {
```

Comment:
Should record come as last argument, within an option object?

Reply:
I'm torn on this one!